typescript/src/lib/kurtosis_context/kurtosis_context.ts

```diff
@@ -0,0 +1,184 @@
+import { EngineServiceClient } from "../../kurtosis_engine_rpc_api_bindings/engine_service_grpc_pb";
+import { LISTEN_PORT} from "../../kurtosis_engine_rpc_api_consts/kurtosis_engine_rpc_api_consts";
+import {EnclaveContext} from "../enclave_context/enclave_context";
+import * as grpc from "grpc";
+import { Result, err, ok } from "neverthrow";
+import {newCreateEnclaveArgs, newDestroyEnclaveArgs} from "../constructor_calls";
+import {
+    CreateEnclaveArgs,
+    CreateEnclaveResponse, DestroyEnclaveArgs, EnclaveAPIContainerInfo, EnclaveAPIContainerStatusMap,
+    EnclaveInfo, GetEnclavesResponse
+} from "../../kurtosis_engine_rpc_api_bindings/engine_service_pb";
+import {APIContainerContext} from "../api_container_context/api_container_context";
+import * as google_protobuf_empty_pb from "google-protobuf/google/protobuf/empty_pb";
+import * as jspb from "google-protobuf";
+
+const LOCAL_HOST_IP_ADDRESS_STR: string = "0.0.0.0";
+
+export class KurtosisContext {
+    private readonly client: EngineServiceClient;
+
+    private constructor(client: EngineServiceClient) {
+        this.client = client;
+    }
+
+    public newKurtosisContextFromLocalEngine(): Result<KurtosisContext, Error>{
+        let kurtosisEngineSocketStr: string = LOCAL_HOST_IP_ADDRESS_STR + LISTEN_PORT;
```

Comment:
Oops, this is a bug because it's missing a `:`. Typescript has format-strings as well that are actually pretty sweet; they use backticks like this:
```
const kurtosisEngineSocketStr: string = `${LOCAL_HOST_IP_ADDRESS_STR}:${LISTEN_PORT}`;
```